### PR TITLE
ICDS Dashboard: Refresh chart on data change

### PR DESCRIPTION
**Pull request by saiprakash-v**
Jira: https://dimagi-dev.atlassian.net/browse/ICDS-988
##### SUMMARY

Before: Initial Height is set to 1500, and after fetching chart data, Chart is rendered first for a height of 1500, and then in callback (which triggers after chart is rendered), height is calculated based on number of horizontal bars, and chart is re-rendered. 
(Everytime there is a change in chart Options or chart data, chart automatically refreshes)

Change in this PR: watching chart data, when there is a change in it, we calculate the height beforehand without waiting for callback. So as soon as data is available, chart renders with proper height.

### CODE REVIEW

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js [hunk 1]
```diff
@@ -116,10 +116,22 @@ function MapOrSectorController($scope, $compile, $location, storageService, loca
         $compile(popup[0])($scope);
     };
 
+    // watching mapData for any changes. As soon as there is change in data, height of chart is calculated and
+    // chart is refreshed.
+    $scope.$watch(function () {
+        return vm.data.mapData;
+    }, function () {
+        if (vm.data.mapData && vm.data.mapData.chart_data) {
+            var chartHeight = vm.data.mapData.chart_data[0].values.length * 70;
```
Comment (czue) [inline custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:125]: why did this change to 70?

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js [hunk 2]
```diff
@@ -116,10 +116,22 @@ function MapOrSectorController($scope, $compile, $location, storageService, loca
...
+            vm.chartOptions.chart.height = ((chartHeight > 280) ? chartHeight : 280);
```
Comment (czue) [inline custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:126]: where did 280 come from?

**Review by czue — COMMENTED** [review]
Looks good but just waiting for answers before approval

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js (hunk 1 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:125]: There were some locations with names that were too long taking 5 lines. Earlier when it was 60, they were overlapping. So I just increased them to 70

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js (hunk 2 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js:126]: We are giving height to the chart based on number of bars, But each bar doesn't take height of 70 each themselves, this was done keeping in mind that the calculated height includes the labels  and padding of the charts etc. So when there are less number of bars, padding and labels take their usual height, and bars get thinner resulting in overlapping location names. So When there are less than 4 horizontal bars, I gave them the height of a chart which is atleast the height of 4 bars one.
(This isnt the correct thing to do. Open for any suggestions)

Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js]: I have given the padding a certain value, which will always be added, irrespective of number of entries in the chart. I hope this is a bit better than earlier.

**Review by czue — APPROVED** [review]

File: custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js (hunk 2 above)
Comment (czue) [reply custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js]: sounds good, thanks

**Pull request merged**